### ISSUE TITLE
Report generation for LGST results object

### ISSUE THREAD

**Issue by AdriaanRol** [pyGSTio/pyGSTi#39]
@kmrudin @enielse 

I am able to run linear gst with on a dataset using `result = pygsti.do_lgst(ds, ... )`. Even for 2 qubit GST this runs in ~2.5 secs, as opposed to the multiple hours it takes me to do the standard practice GST `result = pygsti.do_stdpractice_gst(ds, ...)` . 

However, when I try to create a report from the results object that `do_lgst` gives me this is not possible. Because the result object does not contain the `estimates` attribute (see error messages below). I understand that the report generated by LGST is not to be considered reliable, however it does provide a very valuable sanity check. 

Is there any chance this bug will be addressed in the future or is this behavior that is not supported? 

I'm running on the latest version of the  beta branch:  472a06d58204c97ffed6e99fe71aa0ae22b8e263 . 

```python
pygsti.report.create_standard_report(
    results=result, title=a.measurementstring+'_'+a.timestamp, 
    filename=join(a.proc_data_dict['folder'], a.measurementstring+'_'+a.timestamp +'line_inv_GST_report.html'),
    confidenceLevel=95)
```
```
*** Creating workspace ***

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-13-14479d2824b1> in <module>()
      2     results=result, title=a.measurementstring+'_'+a.timestamp,
      3     filename=join(a.proc_data_dict['folder'], a.measurementstring+'_'+a.timestamp +'line_inv_GST_report.html'),
----> 4     confidenceLevel=95)

~/GitHubRepos/DiCarloLab_Repositories/pyGSTi/packages/pygsti/report/factory.py in create_standard_report(results, filename, title, confidenceLevel, comm, ws, auto_open, link_to, brevity, advancedOptions, verbosity)
    669 
    670     results_dict = results if isinstance(results, dict) else {"unique": results}
--> 671     toggles = _set_toggles(results_dict, brevity, combine_robust)
    672 
    673     #DEBUG

~/GitHubRepos/DiCarloLab_Repositories/pyGSTi/packages/pygsti/report/factory.py in _set_toggles(results_dict, brevity, combine_robust)
    186     toggles["ShowScaling"] = False
    187     for res in results_dict.values():
--> 188         for est in res.estimates.values():
    189             weights = est.parameters.get("weights",None)
    190             if weights is not None and len(weights) > 0:

AttributeError: 'GateSet' object has no attribute 'estimates'

```

**Comment by enielse** [comment]
`pygsti.do_lgst(...)` returns a `GateSet` object, not a `Results` object (see its docstring). The `AttributeError` shown above is due to you passing a `GateSet` to `create_standard_report` where is expects a `Results` object - so the error is not a bug.  You raise a good point, however, which is "how do I just run LGST and generate a report?".  There isn't a good way to do this currently, but I think this would be a great feature/functionality to include.  I'll try (but not promise :) to create a nice way of doing this by the next release, which should only be about 2 weeks away.

The closest thing you can do currently (in version 0.9.4.x) is run `do_stdpractice_gst` with a single maximum-length of 0 and with a large tolerance, so the slow iterative optimization exits quickly.  You should be able to do this using something like (where I've also added `'onBadFit': []`, so it doesn't do anything extra if the fit is bad, and `modes="TP"` so it only does a single optimization) :

```
maxLengths= [0]
global_opts = {'tolerance':1000.0, 'onBadFit': []}
result = pygsti.do_stdpractice_gst(ds, gs_target, prep_fiducials,
                                   meas_fiducials, germs, maxLenghts,
                                   advancedOptions={'all': global_opts},
                                   modes="TP")
```

 Let me know how long this takes.  It will still do more than it needs to, so it won't be 2.5s, but it will  hopefully take significantly less than an hour.

**Comment by AdriaanRol** [comment]
@enielse 
> The closest thing you can do currently (in version 0.9.4.x) is run do_stdpractice_gst with .... 

Thanks for pointing that out, I'll try that today. Looking forward to the inclusion of this new feature.

**Comment by AdriaanRol** [comment]
@enielse I tried the "quick" GST. It seems to work as expected. However, when specifying maxLenghts=[0], an Attribute Error is raised in the report generation. When instead I set maxLenghts=[1] everything works fine. Running this simple version of GST takes ~1.5 mins including report generation and is a huge lifesaver to determine if data makes sense! thanks for the help.

**Comment by enielse** [comment]
Great! I'm glad this basically worked.  Changing the `[0]` to `[1]` as you did is fine.  In addition to not causing an error, using `[1]` will cause the color box plots in the "Per-sequence detail" tab of the report to show something (i.e. not be blank).

**Comment by enielse** [comment]
As of the just-released version 0.9.5, you can generate LGST-only reports even more easily, using the new `do_linear_gst` function like so:
```
results = pygsti.do_linear_gst(ds, gs_target, prep_fiducials, meas_fiducials)
pygsti.report.create_standard_report(results, filename="example_files/LGSTonlyReport",
                                     title="LGST-only Example Report", verbosity=2)
````
This is further detailed in the new example notebook:
[**Reports - How to create an LGST-only report.ipynb**](https://github.com/pyGSTio/pyGSTi/blob/master/jupyter_notebooks/Examples/Reports%20-%20How%20to%20create%20an%20LGST-only%20report.ipynb)

**Comment by enielse** [comment]
Closing since I think this issue is resolved and it's been a while without activity.
